src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -661,7 +697,16 @@ static Value builtinArraySlice(ExecutionState& state, Value thisValue, size_t ar
     uint32_t finalEnd = (relativeEnd < 0) ? std::max((double)len + relativeEnd, 0.0) : std::min(relativeEnd, (double)len);
 
     int64_t n = 0;
+#ifdef ESCARGOT_ENABLE_ES2015
+    // Let count be max(final - k, 0).
+    // Let A be ArraySpeciesCreate(O, count).
+    Value val = arraySpeciesCreate(state, thisObject, std::max(finalEnd - k, (int64_t)0));
+    ASSERT(val.isObject());
+    ASSERT(val.asObject()->isArrayObject());
```

Comment:
For readability / State expression explicitly
ASSERT(val.isObject()->isArrayObject());